qrcode_detection_package/main.py

```diff
@@ -1,56 +1,358 @@
+        elif (self.config_detection_method == 1):
+            # Initialize a VideoCapture object for the camera
+            cap = cv2.VideoCapture(0)
+
+            # To-Do: Evaluate influence of different camera settings to quality and performance on the real hardware
+            cap.set(cv2.CAP_PROP_FRAME_WIDTH, 4056)
+            cap.set(cv2.CAP_PROP_FRAME_HEIGHT, 3040)
+
+            # Check if the VideoCapture object was opened successfully
+            if not cap.isOpened():
+                self.get_logger().info("Error: Unable to open camera")
+
+            # Capture an image from the camera
+            ret, img = cap.read()
+
+            # Check if the image was captured successfully
+            if not ret:
+                self.get_logger().info("Error: Unable to capture image from camera")
+
+            captured_image = img
+
+            # Release the VideoCapture object
+            cap.release()
+
+        elif (self.config_detection_method == 2):
+            try:
+                # set image path with timestamp as name
+                timestamp = time.strftime("%Y-%m-%d_%H-%M-%S")
+                image_path = f"/image/{timestamp}.jpg"
+                # execute libcamera command to capture imagae
+                command = ["libcamera-jpeg", "-o", image_path]
+                subprocess.run(command)
+            except:
+                self.get_logger().info("libcamera could not take picture")
+            captured_image = cv2.imread(image_path)
+
         return captured_image
 
+    def __corners_to_middlepoint(self, points):
+        """
+        Calculate the geometric midpoint of a rectangle formed by 4 points.
+
+        Args: 
+            points: A list of tuples representing the coordinates of the 4 points.
+
+        Returns: 
+            A tuple representing the geometric midpoint of the rectangle.
+        """
+        # Extract the coordinates of the diagonal points
+        top_left = points[0][0]
+        bottom_right = points[0][2]
+
+        # Calculate the geometric midpoint
+        midpoint_x = (top_left[0] + bottom_right[0]) / 2
+        midpoint_y = (top_left[1] + bottom_right[1]) / 2
+
+        # Return the midpoint as a tuple (x, y)
+        return (midpoint_x, midpoint_y)
+
+    def __relative_midpoint(self, midpoint_x, midpoint_y, img_width, img_height):
+        """
+        Calculate the position of the QR code relative to the middle of the image in percent from 
+        -100% to 100% relative to image height and width.
+
+        This function calculates the position of the QR code relative to the middle of the image.
+        The position is represented in percent from -100% to 100% relative to the image height and width.
+        The function takes absolute midpoint coordinates (midpoint_x, midpoint_y) and the image width 
+        and height as input.
+        It returns a tuple representing the relative position of the QR code to the middle of the picture.
```

Comment:
This is duplicate info as params and return are described below. You could leave this out imo.